src/OSPSuite.InstallationValidator.Core/Services/BatchStarterTask.cs

```diff
@@ -25,13 +29,42 @@ public BatchStarterTask(IInstallationValidatorConfiguration applicationConfigura
 
       private string logFilePath(string basePath) => Path.Combine(basePath, Constants.Tools.BATCH_LOG);
 
-      public Task StartBatch(string outputFolderPath, CancellationToken cancellationToken)
+      public async Task<BatchRunSummary> StartBatch(string outputFolderPath, CancellationToken cancellationToken)
       {
-         return Task.Run(() =>
+         var batchRunResult = new BatchRunSummary
+         {
+            ComputerName = Environment.MachineName,
+            StartTime = DateTime.Now,
+            MoBiVersion = moBiVersion(),
+            PKSimVersion = pkSimVersion(),
+            BatchOutputFolder = outputFolderPath,
+            OperatingSystem = Environment.OSVersion,
+            ConfigurationInputFolder = _applicationConfiguration.BatchInputsFolderPath
+         };
+
+         await Task.Run(() =>
          {
             var logFile = logFilePath(outputFolderPath);
             startBatchProcess(outputFolderPath, cancellationToken, logFile);
+            batchRunResult.EndTime = DateTime.Now;
          }, cancellationToken);
+
+         return batchRunResult;
```

Comment:
you can return that from the Task.Run and return directly the Task. No need to await here at all I think